common/common-rest/src/main/java/io/servicecomb/common/rest/RestProducerInvocation.java

```diff
@@ -64,6 +65,18 @@ public void invoke(Transport transport, HttpServletRequestEx requestEx, HttpServ
 
   protected void scheduleInvocation() {
     OperationMeta operationMeta = restOperationMeta.getOperationMeta();
+    try {
+      Object[] args = RestCodec.restToArgs(requestEx, restOperationMeta);
+      this.invocation = InvocationFactory.forProvider(transport.getEndpoint(),
+          restOperationMeta.getOperationMeta(),
+          args);
+      MetricsDataMonitorUtil.prepareStartTimeMetrics(invocation);
```

Comment:
why static call everywhere? it makes metrics implementation coupled with each module in which it's used.

not only unit testing has to involve metrics implementation, but also it makes switching underlying metrics library nearly impossible

Reply:
fixed